src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/PagedBufferedTextWriter.cs

```diff
@@ -89,21 +106,54 @@ public override void Write(string value)
             _charBuffer.Append(value);
         }
 
-        public override async Task WriteAsync(char value)
+        public override Task WriteAsync(char value)
+        {
+            var flushTask = FlushAsyncCore();
+
+            // FlushAsyncCore will return CompletedTask if nothing sync buffered
+            // Fast-path and skip async state-machine if only a single async operation
+            return ReferenceEquals(flushTask, Task.CompletedTask) ? 
```

Comment:
I've usually written this as `Task.IsCompletedSuccessfully ` for clarity - is there a material difference?

Reply:
Doesn't exist in .NET Standard 2.0 and the alternative ways of checking are quite heavy/ugly https://github.com/dotnet/corefx/issues/16745; whereas `ReferenceEquals` is super fast (assuming that's the `Task` instance being returned; though most things use it).

Reply:
Cool beans

Reply:
`IsCompletedSuccessfully` will also include `async` methods that complete sync; but alas, not .NET Standard :cry: